packages/core/src/amazonq/activation.ts

```diff
@@ -52,14 +55,25 @@ export async function activate(context: ExtensionContext) {
         listCodeWhispererCommandsWalkthrough.register(),
         focusAmazonQPanel.register(),
         focusAmazonQPanelKeybinding.register(),
-        tryChatCodeLensCommand.register()
+        tryChatCodeLensCommand.register(),
+        vscode.workspace.onDidChangeConfiguration(async configurationChangeEvent => {
+            if (configurationChangeEvent.affectsConfiguration('amazonQ.localWorkspaceIndex')) {
+                if (CodeWhispererSettings.instance.isLocalIndexEnabled()) {
+                    const setupLsp = debounce(async () => {
+                        void LspController.instance.trySetupLsp(context)
+                    }, 5000)
+                    void setupLsp()
+                }
+            }
+        })
     )
 
     Commands.register('aws.amazonq.learnMore', () => {
         void vscode.env.openExternal(vscode.Uri.parse(amazonQHelpUrl))
     })
 
     await activateBadge()
+    void LspController.instance.trySetupLsp(context)
```

Comment:
This one isn't debounced. Perhaps the debounced function should be declared so that this can also call the debounced function?